docs/developer/development-environment.md

```diff
@@ -0,0 +1,130 @@
+conda activate fedbiomed-dev
+```
+
+
+### Install Pdm
+
+Once virtual environments are set `PDM` package manager can be installed via `PyPI`. It is also possible to instal `PDM` globally for the Python version that is activated or base `conda` environment if `conda` is preferred as virtual environment manager.
+
+```
+pip install pdm
+```
+
+The command above will install Poetry in the active `conda` environment or in active `pyenv` if it is used for managing Python versions.
+
+
+## Using PDM
+
+PDM is a tool to manage dependencies of a Python project, and build the python package from its source through instructions given in `pyproject.toml` located in the root directory of the project. Project build settings and dependencies can be configured form `pyproject.toml` file.
+
+Once PDM is installed dependencies can be installed through the following command. If there is no lock file generated by poetry it will resolve all the dependencies defined in `pyproject.toml`
+
+```
+pdm install
+```
+
+The following command will install all dependency groups except the dependencies that are depended as extra. There are also other installation options such selection of specific groups or extra dependencies. Please see `pdm install --help` and `pyproject.toml` file for more information.
+
+This command will automatically detect the environment that used currently and install all dependencies in it. For example, if conda env is activated and there is no `.venv` folder in root directory of the project, all dependencies will installed within the `conda` environment. However, if another virtual environment is activated, or if `.venv` folder is existing poetry will install dependencies within that environment.
+
+If there is no virtual environment created pdm will create a default virtual environment using python's `virtualenv` in `.venv`
+
+
+### What about instant changes in source code?
+
+Poetry automatically add package source to your environment that makes `fedbiomed` module accessible from Python interpreter. It also handles instant changes in the source code. If a new module is added or a module is edited poetry will automatically make changes available within the environment currently used. It means there is no need to install or rebuild `fedbiomed` package.
+
+!!! warning "Important"
+    You should never install `fedbiomed` from `pip` in your development environment. This my interrupt to access current source code changes in your development environment.
+
+### Add new module
+
+New packages can be added through `pyproject.toml`, or using the command `poetry add`. Please see `peotry add --help` for detailed usage instructions.
+
+Once a new package added via `pyproject.toml` it may be required to run `poetry lock --no-update` to resolve the dependencies, then, execute `poetry install` to update packages and install missing ones.
+
+### For more
+
+Please visit `poetry` documentation for more information and usage details.
+
```

Comment:
Is `poetry` is still used. I think we should remove this line

Reply:
I replaced poetry with pdm. Thanks for the remark